subsys/dfu/src/dfu_target_modem.c

```diff
@@ -195,6 +195,18 @@ int dfu_target_modem_write(const void *const buf, size_t len)
 		} else {
 			return 0;
 		}
+	case DFU_AREA_NOT_BLANK:
```

Comment:
You shouldn't need to check the offset then, the only thing you can do is erase.
It will look pretty much the same as what you are doing for `DFU_INVALID_FILE_OFFSET`, though I don't understand why in that case you are also attempting to write again (recursively?!) since you might be writing something that is not the beginning of the image right after erasing the flash?

Reply:
So this is the issue, I haven't tried it so I'm not sure if this trick works. The problem is that modem does not report `DIRTY_IMAGE` after having done a complete modem FW upgrade with a debugger. It will report `offset == 0` but then report `DFU_AREA_NOT_BLANK` when you try to write to it.

It could be that I'm doing something wrong here I just quickly wrote something I expected to fix a bug. The problem is that on older modem FW I've seen cases where `DFU_AREA_NOT_BLANK` is reported when we are expecting it to report `DFU_ERASE_PENDING` when trying to write to it ending up in a locked state. Where it keeps trying to recursively write as you said.

Reply:
This was also intended to be draft PR.

Reply:
I'm sure I tried to do the same as `DFU_INVALID_FILE_OFFSET` before but would get into a problem where it was locked recursively as you said. Where every write got `DFU_AREA_NOT_BLANK`.

Could be that I'm wrong so I'll try that first.

Reply:
So this works for mfw1.1.0 but I want it to be tested for all release mfw because I vaguely remember that we encountered an issue with doing this in a previous version which would cause the program flow to hang(loop by recursively calling delete then write).